lib/cose/rsapkcs1_algorithm.rb

```diff
@@ -40,4 +40,11 @@ def to_pkey(key)
 COSE::Algorithm.register(RSAPKCS1Algorithm.new(-257, "RS256", hash_function: "SHA256"))
 COSE::Algorithm.register(RSAPKCS1Algorithm.new(-258, "RS384", hash_function: "SHA384"))
 COSE::Algorithm.register(RSAPKCS1Algorithm.new(-259, "RS512", hash_function: "SHA512"))
+
+# Patch openssl-signature_algorithm gem to support discouraged/deprecated RSA-PKCS#1 with SHA-1
+# (RS1 in JOSE/COSE terminology) algorithm needed for WebAuthn.
+OpenSSL::SignatureAlgorithm::RSAPKCS1.const_set(
+  :ACCEPTED_HASH_FUNCTIONS,
+  OpenSSL::SignatureAlgorithm::RSAPKCS1::ACCEPTED_HASH_FUNCTIONS + ["SHA1"]
+)
```

Comment:
I know there must be a reason, but I'm curious: why don't we go ahead and add it in the gem itself?

Reply:
Yeah, great question.

Given SHA-1 been deprecated/discouraged for 10+ years, don't want to promote it's use in `openssl-signature_algorithm`, which can be perfectly used in isolation.

We need to add it here because of WebAuthn backwards compatiblity with `RS1`, so prefer to monkey-patch here instead of forcing other gems to also support it.